protocols/kad/src/query/peers/closest.rs

```diff
@@ -483,7 +483,8 @@ mod tests {
     use rand::{rngs::StdRng, Rng, SeedableRng};
     use std::{iter, time::Duration};
 
-    fn random_peers<R: Rng>(n: usize, g: &mut R) -> Vec<PeerId> {
+    fn random_peers(n: usize) -> Vec<PeerId> {
+        let mut g = rand::thread_rng();
```

Comment:
With this change the `random_peers` function is no longer deterministic over the provided random number generator, thus making debugging the tests below harder.

How about replacing the `g.gen::<[u8; 32>` below, or seeding a `rand::Rng` with a deterministic number from `g: Gen`?